future_test.go

```diff
@@ -233,3 +236,41 @@ func TestFutureGetIteratorError(t *testing.T) {
 		}
 	}
 }
+
+func TestFutureCancel(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	cancel()
+	canceler := mock.NewCancelerMock(func(futA *Future, futB *Future) bool { return futA == futB })
+	fut := NewFuture(ctx, canceler)
+	ok := canceler.AddFuture(fut)
+	if !ok {
+		t.Fatal("fut is not set in the queue")
+	}
+	RefImplFutureWait(fut) // fut.wait() inside
+	err := fut.Err()
+	if err == nil || err != ContextDoneErr {
+		t.Errorf("wrong error catched")
+	}
+
+	if canceler.FindFuture(fut) != nil {
+		t.Errorf("the future is not canceled")
+	}
+}
+
+func TestFuturePushCancel(t *testing.T) {
```

Comment:
The Future interface still not tested well (I can miss something):
1. Cancel a context + Future.Get()
2. Cancel a context + Future.AppendPush()
3. Cancel a context + Future.WaitChan()
4. Cancel a context + Future.SetError()
5. Cancel a context + Future.SetResponse()
6. Context + timeout iterator.